File: Ejercicios/Ejercicio4/4.cpp
#include <iostream>
using namespace std;

// Interfaz para dispositivos de reproduccion de musica
class ReproductorMusica 
{
public:
    virtual void reproducir() = 0;
};

// Interfaz para dispositivos de vista de video
class Televisor 
{
    public:
        virtual void mostrarVideo() = 0;
};

// Clase para un reproductor de música
class ReproductorMP3 : public ReproductorMusica 
{
    public:
        void reproducir() override 
        {
            cout << "Reproduciendo música..." << endl;
        }
};

// Clase para un televisor básico
class TelevisorBasico : public Televisor 
{
public:
    void mostrarVideo() override 
    {
        cout << "Mostrando video..." << endl;
    }
};

int main() 
{
    ReproductorMusica* mp3 = new ReproductorMP3();
    Televisor* tv = new TelevisorBasico();

    mp3->reproducir();
    tv->mostrarVideo();

    delete mp3;
    delete tv;

    return 0;
}

File: Ejercicios/Ejercicio4/4.py
# Interfaz para dispositivos de reproduccion de musica
class ReproductorMusica:
    def reproducir(self):
        pass

# Interfaz para dispositivos de vista de video
class Televisor:
    def mostrarvideo(self):
        pass

# Clase para un reproductor de musica
class ReproductorMP3(ReproductorMusica):
    def reproducir(self):
        print("Reproduciendo musica...")

# Clase para un televisor basico
class TelevisorBasico(Televisor):
    def mostrarvideo(self):
        print("Mostrando video...")


mp3 = ReproductorMP3()
tv = TelevisorBasico()

mp3.reproducir()
tv.mostrarvideo()